providers/dns/alidns/alidns.go

```diff
@@ -25,48 +51,74 @@ type DNSProvider struct {
 func NewDNSProvider() (*DNSProvider, error) {
 	values, err := env.Get("ALICLOUD_ACCESS_KEY", "ALICLOUD_SECRET_KEY")
 	if err != nil {
-		return nil, fmt.Errorf("AliDNS: %v", err)
+		return nil, fmt.Errorf("alicloud: %v", err)
 	}
 
-	regionID := os.Getenv("ALICLOUD_REGION_ID")
+	config := NewDefaultConfig()
+	config.APIKey = values["ALICLOUD_ACCESS_KEY"]
+	config.SecretKey = values["ALICLOUD_SECRET_KEY"]
+	config.RegionID = os.Getenv("ALICLOUD_REGION_ID")
 
-	return NewDNSProviderCredentials(values["ALICLOUD_ACCESS_KEY"], values["ALICLOUD_SECRET_KEY"], regionID)
+	return NewDNSProviderConfig(config)
 }
 
-// NewDNSProviderCredentials uses the supplied credentials to return a DNSProvider instance configured for alidns.
+// NewDNSProviderCredentials uses the supplied credentials
+// to return a DNSProvider instance configured for alidns.
+// Deprecated
 func NewDNSProviderCredentials(apiKey, secretKey, regionID string) (*DNSProvider, error) {
-	if apiKey == "" || secretKey == "" {
-		return nil, fmt.Errorf("AliDNS: credentials missing")
+	config := NewDefaultConfig()
+	config.APIKey = apiKey
+	config.SecretKey = secretKey
+	config.RegionID = regionID
+
+	return NewDNSProviderConfig(config)
+}
+
+// NewDNSProviderConfig return a DNSProvider instance configured for alidns.
+func NewDNSProviderConfig(config *Config) (*DNSProvider, error) {
+	if config == nil {
+		return nil, errors.New("alicloud: the configuration of the DNS provider is nil")
+	}
+
+	if config.APIKey == "" || config.SecretKey == "" {
+		return nil, fmt.Errorf("alicloud: credentials missing")
 	}
 
-	if len(regionID) == 0 {
-		regionID = defaultRegionID
+	if len(config.RegionID) == 0 {
+		config.RegionID = defaultRegionID
 	}
 
-	client, err := alidns.NewClientWithAccessKey(regionID, apiKey, secretKey)
+	conf := sdk.NewConfig().WithTimeout(config.HTTPTimeout)
+	credential := credentials.NewAccessKeyCredential(config.APIKey, config.SecretKey)
+
+	client, err := alidns.NewClientWithOptions(config.RegionID, conf, credential)
 	if err != nil {
-		return nil, fmt.Errorf("AliDNS: credentials failed: %v", err)
+		return nil, fmt.Errorf("alicloud: credentials failed: %v", err)
```

Comment:
Another vote for replacing these with `github.com/pkg/errors.Wrap`?

Reply:
nope, I don't think we need to use `github.com/pkg/error`